packages/headless/src/features/debug/ranking-info-parser.test.ts

```diff
@@ -43,14 +43,14 @@ describe('ranking infor parser', () => {
     const parsed = parseRankingInfo(toParse)!;
     const testsTermsWeights = parsed.termsWeight?.['test'];
 
-    expect(testsTermsWeights?.Weights.Title).toBe(800);
-    expect(testsTermsWeights?.Weights.Concept).toBe(0);
-    expect(testsTermsWeights?.Weights.Summary).toBe(300);
-    expect(testsTermsWeights?.Weights.URI).toBe(500);
-    expect(testsTermsWeights?.Weights.Formatted).toBe(0);
-    expect(testsTermsWeights?.Weights.Casing).toBe(0);
-    expect(testsTermsWeights?.Weights.Relation).toBe(200);
-    expect(testsTermsWeights?.Weights.Frequency).toBe(1744);
+    expect(testsTermsWeights?.Weights?.Title).toBe(800);
+    expect(testsTermsWeights?.Weights?.Concept).toBe(0);
+    expect(testsTermsWeights?.Weights?.Summary).toBe(300);
+    expect(testsTermsWeights?.Weights?.URI).toBe(500);
+    expect(testsTermsWeights?.Weights?.Formatted).toBe(0);
+    expect(testsTermsWeights?.Weights?.Casing).toBe(0);
+    expect(testsTermsWeights?.Weights?.Relation).toBe(200);
+    expect(testsTermsWeights?.Weights?.Frequency).toBe(1744);
```

Comment:
Removing some casts from the `ranking-info-parser` code allowed typescript to detect that `Weights` could be null. This is technically a breaking change. With that said, I'm not sure if this could happen in practice, in that if we have termWeights, we will always be able to parse a `Weights` object?